### ISSUE TITLE
HAN模型中你使用reduce_mean 来替换 论文中的softmax求 attention weight， 这样做的目的是什么吗

### ISSUE THREAD

**Issue by laoyin** [RandolphVI/Multi-Label-Text-Classification#14]
作者您好，我在看您的代码时，发现了您提供的论文链接里，attention 和您的代码实现，发现有出入，按照论文应该是tf.nn.softmax 来求weight，我想问，使用平均值是您进行的调试后 选择的最优方案吗？
###
           # 2. Compute weight by computing similarity of u and attention vector u_attention
            score = tf.multiply(u, u_attention)  # [batch_size, sequence_length, num_units]
            weight = tf.reduce_mean(score, axis=2, keepdims=True)  # [batch_size, sequence_length, 1]

            # 3. Weight sum
            self.attention = tf.reduce_sum(tf.multiply(u, weight), axis=1)  # [batch_size, num_units]

**Comment by RandolphVI** [comment]
是的，这里采用 reduce_mean() 是针对我自己个人数据作出的小调整。

**Comment by RandolphVI** [comment]
@laoyin 
是的，这里采用 reduce_mean() 是针对我自己个人数据作出的小调整。

**Comment by laoyin** [comment]
ok，多谢
